packages/components/src/components/contract-card/contract-card-items/contract-update-form.tsx

```diff
@@ -131,7 +133,10 @@ const ContractUpdateForm = (props: TContractUpdateFormProps) => {
     };
 
     const isStateUnchanged = () => {
-        return isDeepEqual(getStateToCompare(getContractUpdateConfig(contract_info)), getStateToCompare(props));
+        return isDeepEqual(
+            getStateToCompare(getContractUpdateConfig(contract_info)),
+            getStateToCompare(props.contract)
```

Comment:
many thanks to @kate-deriv  who noticed this issue and active participated in finding the solution :)